shell/platform/windows/key_event_handler.cc

```diff
@@ -38,13 +59,49 @@ void KeyEventHandler::KeyboardHook(Win32FlutterWindow* window,
                                    int key,
                                    int scancode,
                                    int action,
-                                   int mods) {
+                                   char32_t character) {
   // TODO: Translate to a cross-platform key code system rather than passing
   // the native key code.
   rapidjson::Document event(rapidjson::kObjectType);
   auto& allocator = event.GetAllocator();
   event.AddMember(kKeyCodeKey, key, allocator);
-  event.AddMember(kKeyMapKey, kAndroidKeyMap, allocator);
+  event.AddMember(kScanCodeKey, scancode, allocator);
+  event.AddMember(kCharacterCodePointKey, character, allocator);
+  event.AddMember(kKeyMapKey, kWindowsKeyMap, allocator);
+
+  // Get all the currently pressed modifiers
+  int mods = 0;
+
+  if (GetKeyState(VK_SHIFT) < 0)
+    mods |= kShift;
+  if (GetAsyncKeyState(VK_LSHIFT) < 0)
+    mods |= kShiftLeft;
+  if (GetKeyState(VK_RSHIFT) < 0)
+    mods |= kShiftRight;
+  if (GetKeyState(VK_CONTROL) < 0)
+    mods |= kControl;
+  if (GetKeyState(VK_LCONTROL) < 0)
+    mods |= kControlLeft;
+  if (GetKeyState(VK_RCONTROL) < 0)
+    mods |= kControlRight;
+  if (GetKeyState(VK_MENU) < 0)
+    mods |= kAlt;
+  if (GetKeyState(VK_LMENU) < 0)
+    mods |= kAltLeft;
+  if (GetKeyState(VK_RMENU) < 0)
+    mods |= kAltRight;
+  if (GetKeyState(VK_LWIN) < 0)
+    mods |= kWinLeft;
+  if (GetKeyState(VK_RWIN) < 0)
+    mods |= kWinRight;
+  if (GetKeyState(VK_CAPITAL) < 0)
+    mods |= kCapsLock;
+  if (GetKeyState(VK_NUMLOCK) < 0)
+    mods |= kNumLock;
+  if (GetKeyState(VK_SCROLL) < 0)
+    mods |= kScrollLock;
```

Comment:
This whole section is calling out for an `int GetModsForKeyState()` helper function (just a free function in the internal namespace you add above) so the details aren't all inline in the higher-level method.